### DISCUSSION TITLE
Mapping hotkey for compile and run in Vim editor

### DISCUSSION THREAD

**Discussion by stefanos82** [c3lang/c3c#1663] [Show and tell]
Those of you who use Vim as your editor of choice and would like to map a hotkey to compile and run C3 projects, you can do so with the following steps:

- Go in `$HOME/.vim/after/ftplugin/`
- Create a folder named `c3`
- In it, create a `.vim` file; I happened to name it `c3mappedkeys.vim`
- Edit it and add `nmap <F5> :up <bar> !clear && /<your-own-location-the-compiler-can-be-found>/c3c/c3c run -q<CR>` and save it

To explain the last point what it actually does, it goes like this: even if you haven't saved your changes, the editor will do so, thanks to the `:up` command (`:up or :update`: Like `:write`, but only write when the buffer has been modified.); then, we clear the terminal **and then** we execute the `compile-and-run but without the verbosity turned on by default` command.

That's really it; enjoy!

**Comment by lerno** [comment]
Should it perhaps be added to the README for the vim editor syntax highlighter?

**Comment by stefanos82** [comment]
Actually it matches the Show and Tell in Discussions.

Can you move it, please? It does not let me :confused:

**Reply by lerno** [reply]
Oh, that actually worked!

**Reply by stefanos82** [reply]
Lovely!

So, if any other ticket can be transferred here to continuous conversation, that would reduce the unnecessary noise we accidentally cause in issues.
